api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -85,6 +89,8 @@
     /**
      * Return the resolved property value with the specified type for the
      * specified property name from the underlying {@link ConfigSource ConfigSources}.
+     * 
+     * Registered converters are applied if they match the propertyType class
```

Comment:
If I have a registered converter for type `List<String>` and the injection point is of type `Collection<? extends String>`, is this considered a "match" in the language of this specification?

Reply:
The only thing that the spec says about this is "The Type parameter of the interface is the target
type the String is converted to". I admit this isn't clear enough because if the target type is List at can also be assigned to a Collection. That would work for programmatic API, but for injection it's way too complicated to support because a producer would have to be created for all supertypes and interfaces.

We'll clarify that for injection, only the exact target type can be injected. Injection of supertypes like `Collection` isn't portable.

Reply:
OK.

In an alternate universe 😄 you could say this in the general specification:

* A `Converter<T>` must be registered for all type conversion.
* `T` must be the _maximally concrete type_ for the conversion.  _Maximally concrete_ means something like `HashSet<String>`, not `Collection<String>`; i.e. the "leaf" of the type hierarchy.  If a `T` were, say, `Collection<T>`, then only `Collection<T>` and `Collection<? extends T>` injection points would work.  If `T` were, by contract, `HashSet<String>`, then all of its transitive type closure injection points would work.

Then to implement CDI injection, you don't scan for injection points, you scan maximally concrete conversion types and blindly register a producer for each of them.  Problem solved for all possible injection or programmatic lookup.

But since that's not how things are written, and we have to scan for injection points, I think your clarifications are fine.  As I noted in the Google Group, there are enough limitations with the specification and CDI integration that I'd err on the side of talking explicitly about what is supported, and say that if there's anything else someone's thinking about, point out that it is undefined.

Reply:
This would be more of a recommendation than a specification. I think that implementations don't have to scan injection points. They can also choose to scan converters and classpath for all classes with implicit conversion. It's up to the implementation.

However, it makes sense to add hints into the spec for implementers even if they are not necessary to specify behavior. I think that JSRs even have a separate document for implementers for that purpose.

Reply:
How would an implementation scan the classpath for "all classes with implicit conversion"?  Since it cannot, I don't think this is a case where the implementation could use that strategy.  That cuts off a whole tree of possible behavior, so it may make sense to enumerate what is explicitly defined behavior and what is not.

Anyway, some clarifications are better than none so I wouldn't hold this PR up for all possible fixes.

Reply:
Application servers can easily scan all the classpath because it's limited to the deployed WAR or EAR. And, in fact, most of them already do. It's definitely up to the implementation which approach they choose.

Reply:
OK.